### PR TITLE
[Feat] Validation of Secrets pre-created in Application Namespace

### PR DESCRIPTION
**Pull request by rahulchheda**
**What this PR does / why we need it**:
- This PR, will add the ability to validate Secrets, in the application namespace, and add those into the Job created.
- Unlike ConfigMap validation, it will not create the secrets, on its own.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
https://github.com/litmuschaos/chaos-executor/issues/13
**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Comment by Sanjay1611** [comment]
Commits from your previous PR are also being shown. Please correct it

File: pkg/utils/configMapUtils.go [hunk 1]
```diff
@@ -62,6 +62,41 @@ func ValidateConfigMaps(configMaps []v1alpha1.ConfigMap, engineDetails EngineDet
...
+			e := errors.New("Aborting Execution, Secret Name or mountPath is invalid")
+			errorList = append(errorList, e)
+			return nil, errorList
+		}
+
+		_, err := clients.KubeClient.CoreV1().Secrets(engineDetails.AppNamespace).Get(v.Name, metav1.GetOptions{})
+
+		if err != nil {
+			//errors = append(errors, err)
+			log.Infof("Unable to find ConfigMap with Name: %v", v.Name)
```
Comment (ksatchit) [inline pkg/utils/configMapUtils.go]: :) I guess you mean secrets

Comment (ksatchit) [reply pkg/utils/configMapUtils.go]: But why is it here in the first place? Not in secretUtils.go?

File: pkg/utils/configMapUtils.go [hunk 2]
```diff
@@ -116,32 +171,8 @@ func CreateConfigMaps(configMaps []v1alpha1.ConfigMap, engineDetails EngineDetai
...
-		volume.ConfigMap.DefaultMode = &k
-		volumesList = append(volumesList, volume)
-		//volumesList[i] = volume
-	}
-	return volumesList
-}*/
-
 // CreateVolumeBuilder build Volume needed in execution of experiments
-func CreateVolumeBuilder(configMaps []v1alpha1.ConfigMap) []*volume.Builder {
+func CreateVolumeBuilder(configMaps []v1alpha1.ConfigMap, secrets []v1alpha1.Secret) []*volume.Builder {
```
Comment (Sanjay1611) [inline pkg/utils/configMapUtils.go]: Do you need this secrets here??

File: pkg/utils/configMapUtils.go [hunk 3]
```diff
@@ -77,6 +112,26 @@ func CheckConfigMaps(engineDetails EngineDetails, config *rest.Config, experimen
 	}
 }
 
+// CheckSecrets checks for the configMaps embedded inside the chaosExperiments
```
Comment (Sanjay1611) [inline pkg/utils/configMapUtils.go]: comment and file name says `configMaps`

File: pkg/utils/configMapUtils.go [hunk 4]
```diff
@@ -157,7 +188,7 @@ func CreateVolumeBuilder(configMaps []v1alpha1.ConfigMap) []*volume.Builder {
 }
 
 // CreateVolumeMounts mounts Volume needed in execution of experiments
-func CreateVolumeMounts(configMaps []v1alpha1.ConfigMap) []corev1.VolumeMount {
+func CreateVolumeMounts(configMaps []v1alpha1.ConfigMap, secrets []v1alpha1.Secret) []corev1.VolumeMount {
```
Comment (Sanjay1611) [inline pkg/utils/configMapUtils.go]: ^^

File: pkg/utils/configMapUtils.go (hunk 2 above)
Comment (ksatchit) [inline pkg/utils/configMapUtils.go]: where are we using the `secrets` inside this?

File: pkg/utils/configMapUtils.go (hunk 4 above)
Comment (ksatchit) [inline pkg/utils/configMapUtils.go]: where do we have mounts for `secrets`?

File: pkg/utils/configMapUtils.go (hunk 2 above)
Comment (rahulchheda) [reply pkg/utils/configMapUtils.go]: Yes, volume has to be built for secrets

File: pkg/utils/configMapUtils.go (hunk 1 above)
Comment (rahulchheda) [reply pkg/utils/configMapUtils.go]: done! The thing was some common utils as volumeBuilders, or volumeMounts, that was keeping them in same file

File: bin/exec.go [hunk 5]
```diff
@@ -119,14 +119,37 @@ func main() {
...
+		} else {
+			log.Infoln("Unable to find Secrets")
+		}
+
 		log.Infof("Printing Validated ConfigMaps: %v", validatedConfigMaps)
+		log.Infof("Printing Validated Secrets: %v", validatedSecrets)
 
 		// 1. []*volume.Builder
-		volumeBuilders := utils.CreateVolumeBuilder(validatedConfigMaps)
+		volumeBuilders := utils.CreateVolumeBuilder(validatedConfigMaps, validatedSecrets)
```
Comment (ksatchit) [inline bin/exec.go]: Sorry to pick again: I understand that CreateVolumeBuilder is a common fn needed for both configmaps and secrets. So why is it inside configMap utils?

File: pkg/utils/configMapUtils.go (hunk 2 above)
Comment (ksatchit) [reply pkg/utils/configMapUtils.go]: Maybe in future (BCH) we have separate createConfigMapBuilder and createSecretBuilder. Just thinking aloud - what if more type of volumes are to be supported.

File: pkg/utils/secretsUtils.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+			return nil, errorList
+		}
+
+		_, err := clients.KubeClient.CoreV1().Secrets(engineDetails.AppNamespace).Get(v.Name, metav1.GetOptions{})
+
+		if err != nil {
+			//errors = append(errors, err)
+			log.Infof("Unable to find Secrets with Name: %v", v.Name)
+
+			log.Infof("Did'nt find the configMap with Name: %v, and Data is also empty. Aborting Execution", v.Name)
```
Comment (ksatchit) [inline pkg/utils/secretsUtils.go]: ConfigMaps? Data?

File: bin/exec.go (hunk 5 above)
Comment (rahulchheda) [reply bin/exec.go]: Yes, so where should I place it? Will place in some commonUtils!

**Comment by Sanjay1611** [comment]
Can you please look into the commits....It's showing all the previous commits that should not be there

**Comment by Sanjay1611** [comment]
Still the problem persists. I think you have not rebased it properly.

File: bin/exec.go [hunk 7]
```diff
@@ -119,14 +119,36 @@ func main() {
 			log.Infoln("Unable to find ConfigMaps")
 		}
 
+		secretsExist, secrets := utils.CheckSecrets(engineDetails, config, engineDetails.Experiments[i])
+
+		var validatedSecrets []v1alpha1.Secret
+		var errorsListForSecrets []error
```
Comment (ksatchit) [inline bin/exec.go]: may be used in future, right now we return upon encountering an error

**Pull request merged**